lib/src/migrators/division.dart

```diff
@@ -116,6 +100,65 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
         isDivisionAllowed: true);
   }
 
+  /// Visits a `/` operation [node] and migrates it to either the `division`
+  /// function or the `slash-list` function.
+  ///
+  /// If [surroundingParens] is provided, this will reuse the existing parens
+  /// instead of creating new ones.
+  void _visitSlashOperation(BinaryOperationExpression node,
+      {ParenthesizedExpression surroundingParens}) {
+    if ((!_isDivisionAllowed && _onlySlash(node)) ||
+        _isDefinitelyNotNumber(node)) {
+      // Definitely not division
+      if (surroundingParens != null) {
+        addPatch(patchBefore(surroundingParens, "slash-list"));
+      } else {
+        addPatch(patchBefore(node, "slash-list("));
+        addPatch(patchAfter(node, ")"));
+      }
+      _visitSlashListArguments(node);
+    } else if (_expectsNumericResult || _isDefinitelyNumber(node)) {
```

Comment:
Consider making `_isDefinitelyNumber()` return `false` for ambiguous binary operations and adding `|| !isPessimistic` to this condition. That way `_isDefinitelyNumber()` returns a result that matches its name regardless of context, and it's more immediately clear when reading this method that a warning will never be emitted in optimistic mode.